### PR TITLE
Use Redux in React components.

### PR DESCRIPTION
**Pull request by developerwaleed**
## Project Requirements

- First of all - make sure that you keep state immutable in reducers! ✅
- Secondly, change the default state in your books reducer from an empty array to an array with a few books. One of the goals is to display them in your React components.✅
- Import Redux Provider and your store in the main component of your React app.✅
- Connect your app to the Redux store.✅
- Make sure that list of books displays books from Redux store.✅
- Use useSelector to consume the state.✅
- In your React component responsible for adding new books, add functionality for "Add new" button click:✅
- Import Redux useDispatch.✅
- Set the data inputs in the local React state (set title and author - NOTE: categories and comments are NOT part of this step)✅.
- Once your new book object is ready to be submitted to Redux store, dispatch a corresponding action (import Action Creator that you created before and use it here).✅
- In your React component responsible for removing books, add a functionality for "Remove" button click:
- Import Redux useDispatch.✅
- Dispatch a corresponding action (import Action Creator that you created before and use it here).✅
- In your React component responsible for categories, add a functionality for "Check status" button click:✅
- Import Redux useDispatch.✅
- Dispatch a corresponding action (import Action Creator that you created before and use it here).✅

### CODE REVIEW

File: src/components/AddBook/AddBook.js [hunk 1]
```diff
@@ -1,18 +1,66 @@
...
+            onChange={handleChangeinput}
+          />
+          <button
+            type="button"
+            className={style.btnAddBook}
+            onClick={handleSubmitbtn}
+          >
+            ADD BOOK
+          </button>
+        </form>
```
Comment (thecodechaser) [inline src/components/AddBook/AddBook.js]: - [ ] I'm getting values true on the add book form and I'm not able to edit these values. Please fix this issue and make sure the field values are empty and users are able to add book data into fields.
 - I would suggest using two-state for these fields and on the input `value` attribute place the state value and change the state on the change event.

image

File: src/components/Book/Book.js [hunk 2]
```diff
@@ -14,15 +22,25 @@ const Book = (bookInfo) => {
...
           </button>
           <div className={style.divider} />
-          <button type="button" className={`${style.btnTextStyle} ${style.TextStyle6}`}>
+          <button
+            type="button"
+            className={`${style.btnTextStyle} ${style.TextStyle6}`}
+            onClick={handleRemovebtn}
+          >
             Remove
           </button>
```
Comment (thecodechaser) [inline src/components/Book/Book.js:37]: - [x] The books from the reducer are being removed when I click on the `remove` button but If I add a new book from the form then I'm not able to remove that book. Please make sure to fix this issue.

**Review by thecodechaser — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ 

## Hi @caasperr👋, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight 🎉
- No linter errors. :heavy_check_mark:
- Gitflow is followed. :heavy_check_mark:
- Professional README file. :heavy_check_mark:
- PR has a good title and summary. ✔️



## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/components/AddBook/AddBook.js (hunk 1 above)
Comment (developerwaleed) [reply src/components/AddBook/AddBook.js]: hi @thecodechaser, i just read your suggestion can you please elaborate the second point as I cannot understand it please thankyou :)

Comment (thecodechaser) [reply src/components/AddBook/AddBook.js]: @caasperr Hi, 
`const [title, setTitle] = useState('')`

` value={title} onChange={(e) => setTitle(e.target.value)}`

You can implement the same for the author.

Comment (developerwaleed) [reply src/components/AddBook/AddBook.js]: @thecodechaser, Thank you so much for elaborating. but I still have a question why use 2 states when we can handle both with objects in 1line? why add 3-4 extra?

Comment (thecodechaser) [reply src/components/AddBook/AddBook.js]: @caasperr That's only a suggestion 😄

**Review by uwadonat — APPROVED** [review]
## Hi @caasperr, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
## highlights
- Professional readme file ✔️
- No linter error ✔️
- Good design ✔
approval
️

# Status: Approved ✔️✔️✔️

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100%.
Please, remember to tag me @uwadonat in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Pull request merged**